### PR TITLE
[DAWN] webpack: Use ES6 and simplify config

### PR DESCRIPTION
**Pull request by kevinji**
This also removes the redundant production file by combining them into one file.

### CODE REVIEW

File: dawn/webpack.production.config.js [hunk 1]
```diff
@@ -1,62 +0,0 @@
...
-const plugins = [
-  new webpack.DefinePlugin({
-    VERSION: JSON.stringify(require('./package.json').version)
-  }),
-  new webpack.DefinePlugin({
-    "global.GENTLY": false
-  }),
-  new webpack.DefinePlugin({
-    'process.env': {
-      'NODE_ENV': '"production"'
```
Comment (kevinmasd) [inline dawn/webpack.production.config.js:34]: We keep a production variable to differentiate what is being developed vs what should be shipped to students

**Review by kevinmasd — CHANGES_REQUESTED** [review]
I would leave off handling Webpack. This can be done later when we can upgrade from Webpack 1 to 2.

File: dawn/webpack.production.config.js (hunk 1 above)
Comment (kevinmasd) [reply dawn/webpack.production.config.js:34]: Actually, after seeing package.json, I'll get back to you on this.

**Comment by kevinji** [comment]
I don't think I've changed any of the code of the webpack file, just ES6-ifyed it. Updating to Webpack 2.0 shouldn't really be affected imo.

**Comment by kevinji** [comment]
@kevinmasd I think this is mergeable without worrying about Webpack 2.0; you should be able to make that upgrade right on top of these changes.

File: dawn/webpack.config.babel.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import path from 'path';
+import webpack from 'webpack';
+import { version } from './package.json';
+
+const target = process.env.NODE_ENV === 'production' ? 'electron' : 'atom';
```
Comment (kevinmasd) [inline dawn/webpack.config.babel.js]: This was a mistake in the originals; we should probably default to electron.

**Review by kevinmasd — CHANGES_REQUESTED** [review]

File: dawn/webpack.config.babel.js (hunk 2 above)
Comment (kevinji) [reply dawn/webpack.config.babel.js]: So should this always be `'electron'`?

Comment (kevinmasd) [reply dawn/webpack.config.babel.js]: Yes

Comment (kevinji) [reply dawn/webpack.config.babel.js]: Fixed in latest commit.

**Review by kevinmasd — APPROVED** [review]

**Pull request merged**